src/cpp/shared/game/game.cpp

```diff
@@ -185,31 +185,27 @@ void philAI::RedistributeTroops(armyGroup *army1, armyGroup *army2, int a1, int
 }
 
 void game::SetRandomHeroArmies(int heroIdx, int isAI) {
-  int randomUpperBound;
-  int randomLowerBound;
+  double randomUpperBound;
+  double randomLowerBound;
   armyGroup *heroArmy = &gpGame->heroes[heroIdx].army;
   bool hasTier[2];
   const int TIER_ONE = 0;
   const int TIER_TWO = 1;
   int tierChance;
+  randomHeroCreatureInfo* creatureFaction = randomHeroArmyBounds[gpGame->heroes[heroIdx].factionID];
 
   hasTier[TIER_ONE] = 1;
   hasTier[TIER_TWO] = Random(0, 99) < 87;
 
   this->ClearArmy(heroArmy);
   for (int creatureTier = 0; creatureTier < 2; ++creatureTier) {
     if (hasTier[creatureTier]) {
-      randomLowerBound = 10 * randomHeroArmyBounds[gpGame->heroes[heroIdx].factionID][creatureTier].lowQuantity;
-      randomUpperBound = 10 * randomHeroArmyBounds[gpGame->heroes[heroIdx].factionID][creatureTier].highQuantity + 9;
+      randomLowerBound = creatureFaction[creatureTier].lowQuantity;
```

Comment:
Did you already pre-multiply these constants by 10?

Reply:
I'm not sure I understand what you are asking.

Reply:
Why'd you remove  the 10*?

Reply:
I thought we spoke about this earlier in Skype. Everything was multiplied by 10 and then divided by 10. I changed the variables to "double", so the constants weren't really needed for rounding anymore.